app/controllers/user_projects_controller.rb

```diff
@@ -0,0 +1,33 @@
+class UserProjectsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_project
+  before_action :set_user
+
+  def create
+    if current_user.managed_projects.include?(@project)
+      user_project = UserProject.create!(user: @user, project: @project)
+      redirect_to @project, notice: "User successfully assigned."
+    else
+      redirect_to projects_url, notice: "You cannot add anyone to project if you did not create it."
+    end
+  end
+
+  def destroy
+    if current_user.managed_projects.include?(@project)
+      user_project = UserProject.find_by(user: @user, project: @project)
+      user_project.destroy
+      redirect_to @project, notice: "User assignment was removed."
+    else
+      redirect_to projects_url, notice: "You cannot remove anyone from project if you did not create it."
+    end
+  end
+
+  private
+  def set_project
+    @project = Project.find_by_id(params[:project_id])
```

Comment:
You can change it to `find` instead of `find_by_id`. It will use id by default and result in 404 when record is not found in production mode.